src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/PosixSignalRegistrationTests.Unix.cs

```diff
@@ -2,54 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Xunit;
-using System.IO;
 using System.Threading;
-using System.Threading.Tasks;
 using System.Runtime.InteropServices;
 using System.Runtime.CompilerServices;
 using Microsoft.DotNet.RemoteExecutor;
+using System.Collections.Generic;
 
 namespace System.Tests
 {
-    public class PosixSignalRegistrationTests
+    public partial class PosixSignalRegistrationTests
     {
-        private static TimeSpan Timeout => TimeSpan.FromSeconds(30);
-
-        [Fact]
-        public void HandlerNullThrows()
+        public static IEnumerable<object[]> UninstallableSignals()
         {
-            Assert.Throws<ArgumentNullException>(() => PosixSignalRegistration.Create(PosixSignal.SIGCONT, null));
+            yield return new object[] { (PosixSignal)9 };
         }
 
-        [Theory]
-        [InlineData(0)]
-        [InlineData(-1000)]
-        [InlineData(1000)]
-        public void InvalidSignalValueThrows(int value)
+        public static IEnumerable<object[]> SupportedSignals()
         {
-            Assert.Throws<ArgumentOutOfRangeException>(() => PosixSignalRegistration.Create((PosixSignal)value, ctx => { }));
+            foreach (PosixSignal value in Enum.GetValues(typeof(PosixSignal)))
+                yield return new object[] { value };
         }
 
-        [Theory]
-        [InlineData((PosixSignal)9)]  // SIGKILL
-        public void UninstallableSignalsThrow(PosixSignal signal)
+        public static IEnumerable<object[]> UnsupportedSignals()
         {
-            Assert.Throws<IOException>(() => PosixSignalRegistration.Create(signal, ctx => { }));
+            yield return new object[] { 0 };
+            yield return new object[] { 3 };
```

Comment:
Why is `3` unsupported?

Reply:
It's a mistake; I copied from the wrong list.  Will fix.